internal/moreos/moreos_func-e_test.go

```diff
@@ -45,68 +46,106 @@ var (
 	moreosSrcDir embed.FS
 )
 
-// TestProcessGroupAttr_Kill tests sending SIGKILL to fake func-e.
-// On linux, we propagate SIGKILL to the child process as the configured SysProcAttr.Pdeathsig
-// in proc_linux.go.
-func TestProcessGroupAttr_Kill(t *testing.T) {
-	// This works only for linux, sending kill -9 on darwin will not kill the process, we need to kill
-	// via pgid or kill the child first.
-	if runtime.GOOS != "linux" {
-		t.Skip()
+// Test_CallSignal tests sending signals to fake func-e.
+func Test_CallSignal(t *testing.T) {
+	type testCase struct {
+		name           string
+		signal         func(*os.Process) error
+		skip           bool
+		waitForExiting bool
 	}
-	tempDir := t.TempDir()
 
-	// Build a fake envoy and pass the ENV hint so that fake func-e uses it
-	fakeEnvoy := filepath.Join(tempDir, "envoy"+Exe)
-	fakebinary.RequireFakeEnvoy(t, fakeEnvoy)
-	t.Setenv("ENVOY_PATH", fakeEnvoy)
-
-	fakeFuncE := filepath.Join(tempDir, "func-e"+Exe)
-	requireFakeFuncE(t, fakeFuncE)
-
-	stderr := new(bytes.Buffer)
-	stdout := new(bytes.Buffer)
-
-	// With an arg so fakeFuncE runs fakeEnvoy as its child and doesn't exit.
-	arg := string(version.LastKnownEnvoy)
-	cmd := exec.Command(fakeFuncE, "run", arg, "-c")
-	cmd.Stdout = stdout
-	cmd.Stderr = stderr
-
-	require.NoError(t, cmd.Start())
-
-	// Block until we reach an expected line or timeout.
-	reader := bufio.NewReader(stderr)
-	waitFor := "initializing epoch 0"
-	require.Eventually(t, func() bool {
-		b, _ := reader.Peek(512) // the error value is always EOF.
-		return strings.HasPrefix(string(b), waitFor)
-	}, 5*time.Second, 100*time.Millisecond, "timeout waiting for stderr to contain %q", waitFor)
-
-	require.Equal(t, Sprintf("starting: %s %s -c\n", fakeEnvoy, arg), stdout.String())
-
-	fakeFuncEProcess, err := process.NewProcess(int32(cmd.Process.Pid))
-	require.NoError(t, err)
-
-	// Get all fake func-e children processes.
-	children, err := fakeFuncEProcess.Children()
-	require.NoError(t, err)
-	require.Equal(t, len(children), 1) // Should have only one child process i.e. the fake envoy process.
-	fakeEnvoyProcess := &os.Process{Pid: int(children[0].Pid)}
-
-	// Kill the fake func-e process.
-	require.NoError(t, cmd.Process.Kill())
-
-	// Wait for the process to die; this could error due to the kill signal.
-	cmd.Wait() //nolint
-
-	// Wait and check if fake func-e and envoy processes are killed.
-	requireFindProcessError(t, cmd.Process, process.ErrorProcessNotRunning)
-	requireFindProcessError(t, fakeEnvoyProcess, process.ErrorProcessNotRunning)
+	tests := []testCase{
+		{
+			name:           "interrupt",
+			signal:         Interrupt,
+			waitForExiting: true,
+			skip:           runtime.GOOS == OSWindows,
```

Comment:
Seems like calling interrupt on CI to a running process gives Windows CI on GHA a hard time. exit code -1073741510. I'll spin up my local windows and see what I can do.